django/http/response.py

```diff
@@ -397,14 +400,44 @@ def __init__(self, *args, as_attachment=False, filename='', **kwargs):
         self.filename = filename
         super().__init__(*args, **kwargs)
 
+    def _wrap_file_to_stream_close(self, filelike):
+        """
+        Replace the given file-like object's `close()` method with a wrapped
+        version that calls `FileResponse.close()`.
+        """
+        if hasattr(filelike, 'close'):
+            filelike_close = filelike.close
+        else:
+            def filelike_close():
+                pass
+
+        class State:
+            closing = False
+
+        def file_wrapper_close():
+            # Prevent an infinite loop since FileResponse.close() tries
+            # to close the objects in self._closable_objects.
+            if State.closing:
+                return
+
+            State.closing = True
+
+            filelike_close()
+            self.close()
+
+        filelike.close = file_wrapper_close
+
     def _set_streaming_content(self, value):
         if not hasattr(value, 'read'):
             self.file_to_stream = None
             return super()._set_streaming_content(value)
 
-        self.file_to_stream = filelike = value
+        filelike = value
+        # Add to _closable_objects before wrapping with close().
```

Comment:
Am not sure all this is necessary as the same object is referenced... Just add the one line to call the method that wraps `close()`.

Reply:
I don’t understand what you’re suggesting here. What are you referring to when you say “all this”?

Reply:
Oh, if you’re just saying to leave the first line alone I can do that.

Reply:
That, but also the comment is irrelevant and should be removed because the new method is not returning a different `filelike` object, but modifying the definition of a method on it.

That means you could call the method before adding to `_closable_objects` and it would work the same.

Reply:
The point is that we don't want to add `filelike` to `_closable_objects` if the original `filelike` didn't have a `close()` method. `_wrap_file_to_stream_close()` will set a `close()` method in either case.

Reply:
Ah, ok. I think I misunderstood the comment.

Reply:
I'll make the comment clearer.